llama_index/indices/base.py

```diff
@@ -56,15 +56,17 @@ def __init__(
                 raise ValueError("nodes must be a list of Node objects.")
 
         self._service_context = service_context or ServiceContext.from_defaults()
-        self._storage_context = storage_context or StorageContext.from_defaults()
-        self._docstore = self._storage_context.docstore
-        self._vector_store = self._storage_context.vector_store
-
-        if index_struct is None:
-            assert nodes is not None
-            index_struct = self.build_index_from_nodes(nodes)
-        self._index_struct = index_struct
-        self._storage_context.index_store.add_index_struct(self._index_struct)
+
+        with self._service_context.callback_manager.as_trace("index_construction"):
+            self._storage_context = storage_context or StorageContext.from_defaults()
+            self._docstore = self._storage_context.docstore
+            self._vector_store = self._storage_context.vector_store
```

Comment:
curious: why do we have these calls in the scope of the `as_trace`? Is it because we expect to do more computation here in the future?

Reply:
Yea, was just trying to cast a wide net basically. Could definitely move it out of scope if the net is too wide haha

Reply:
yea let's move it out for now.

Reply:
Done!